File: waimai/api/src/main/java/club/yunzhi/log/entity/Comment.java
package club.yunzhi.log.entity;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.sql.Timestamp;

@Getter
@Setter
@Entity
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @OneToOne
    private MyOrder myOrder;

    private String restaurantDetail;

    private Long restaurantScore;

    private String riderDetail;

    private Long riderScore;

    @CreationTimestamp
    private Timestamp createTime;
}

File: web/src/entity/comment.ts
import {MyOrder} from './my-order';

/**
 * 评论
 */
export class Comment {

  id?: number;

  myOrder?: MyOrder;

  restaurantScore?: number;

  restaurantDetail?: string;

  riderScore?: number;

  riderDetail?: string;

  createTime?: number;

  constructor(data = {} as {
    id?: number,
    myOrder?: MyOrder,
    restaurantScore?: number,
    restaurantDetail?: string,
    riderScore?: number,
    riderDetail?: string,
    createTime?: number

  }) {
    this.id = data.id;
    this.myOrder = data.myOrder;
    this.restaurantDetail = data.restaurantDetail;
    this.restaurantScore = data.restaurantScore;
    this.riderScore = data.riderScore;
    this.riderDetail = data.riderDetail;
    this.createTime = data.createTime;
  }
}
